soc/bflb/common/e24/cache.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include <bflb_soc.h>
+#include <l1c_reg.h>
+
+void __ramfunc soc_bflb_cache_invalidate(void)
+{
+	uint32_t tmp;
+
+	tmp = *((uint32_t *)(L1C_BASE + L1C_CONFIG_OFFSET));
```

Comment:
I once got the compiler optimizing away a raw memory write when the `volatile` keyword was not added:

`tmp = *((volatile uint32_t *)(L1C_BASE + L1C_CONFIG_OFFSET));`

Or using `<zephyr/sys/sys_io.h>` which has the `volatile`:

`tmp = sys_read32(L1C_BASE + L1C_CONFIG_OFFSET);`

It does not seem to have any downside anywhere here though...

Reply:
sys_read can not be used here as it is in flash. Otherwise I havent had that issue yet but i guess this is a good safety?

Reply:
done